src/al-sap/al_service_utils.cpp

```diff
@@ -64,3 +64,21 @@ bool areMacsEqual(const std::array<uint8_t, 6> &first, const std::array<uint8_t,
 {
     return std::equal(begin(first), end(first), begin(second));
 }
+
+std::vector<unsigned char> convert_u32_into_bytes(uint32_t number)
+{
+    return std::vector<unsigned char>{static_cast<unsigned char>(number >> 24),
+                                      static_cast<unsigned char>(number >> 16),
+                                      static_cast<unsigned char>(number >> 8),
+                                      static_cast<unsigned char>(number >> 0)};
+}
```

Comment:
I think this would be a better alternative that doesn't require bit shifts, which would be more readable for others looking at the codebase.
```suggestion
std::vector<unsigned char> convert_u32_into_bytes(uint32_t number)
{
    uint8_t* bytes = reinterpret_cast<uint8_t*>(&value);
    return vec(bytes, bytes + sizeof(uint32_t));
}
```

Reply:
This change was not working as ieee1905 is expecting in the specific order, so stayed with the old implementation